src/main/scala/org/broadinstitute/dsde/firecloud/trial/ProjectManager.scala

```diff
@@ -0,0 +1,153 @@
+            context.system.scheduler.scheduleOnce(verifyDelay, self, Verify(projectName))
+          } else {
+            logger.warn(s"rawls reports error when creating <$projectName>.")
+          }
+        }
+        // schedule the next creation
+        context.system.scheduler.scheduleOnce(createDelay, self, Create(current+1, target))
+      }
+    }
+  }
+
+  private def verify(projectName: String) = {
+    logger.info(s"verifying project <$projectName> ...")
+    val rawlsName = RawlsBillingProjectName(projectName)
+    val saToken:WithAccessToken = AccessToken(OAuth2BearerToken(HttpGoogleServicesDAO.getTrialBillingManagerAccessToken))
+    // as the trial billing manager, get the list of all projects I own (no API to get a single project)
+    rawlsDAO.getProjects(saToken) map { projects =>
+      projects.find(p => p.projectName == rawlsName) match {
+        case None => // project doesn't exist in rawls. This shouldn't happen.
+          logger.warn(s"Project <$projectName> was created in rawls but not returned by rawls. Calling it an error.")
+          trialDAO.setProjectRecordVerified(rawlsName, verified = true, status = CreationStatuses.Error)
+        case Some(proj) =>
+          proj.creationStatus match {
+            case CreationStatuses.Creating =>
+              logger.info(s"Project <$projectName> still creating; will retry verification.")
+              context.system.scheduler.scheduleOnce(verifyDelay, self, Verify(projectName))
+            case CreationStatuses.Ready =>
+              logger.info(s"Project <$projectName> verified as ready!")
+              trialDAO.setProjectRecordVerified(rawlsName, verified = true, status = CreationStatuses.Ready)
+            case CreationStatuses.Error =>
+              logger.warn(s"Project <$projectName> errored during creation: ${proj.message}")
+              trialDAO.setProjectRecordVerified(rawlsName, verified = true, status = CreationStatuses.Error)
+          }
+      }
+    } recover {
+      case t:Throwable =>
+        logger.warn(s"Error checking status on <$projectName>: ${t.getMessage}")
+        context.system.scheduler.scheduleOnce(verifyDelay, self, Verify(projectName))
+    }
+  }
+
+  private def getAndRecordUniqueProjectName: Option[String] = {
+    def verifyUniqueProjectName(name: String, currentAttempt: Int): Option[String] = {
+      val sanityLimit = 100
+      Try(trialDAO.insertProjectRecord(RawlsBillingProjectName(name))) match {
+        case scala.util.Success(project) =>
+          logger.debug(s"found unique project name in $currentAttempt attempt(s); recorded in pool.")
+          Some(project.name.value)
+        case scala.util.Failure(f) =>
+          if (currentAttempt > sanityLimit) {
+            logger.error(s"Could not generate a unique project name after $currentAttempt tries: ${f.getMessage}")
+            None
+          } else {
+            verifyUniqueProjectName(ProjectNamer.randomName, currentAttempt+1)
+          }
```

Comment:
do we need any sort of delay before we try again?

Reply:
I don't think this needs a retry. But, I changed the default refresh policy for the dao, so that at runtime we ensure updates - from any user - are committed before trying again.